torch/_inductor/choices.py

```diff
@@ -220,13 +249,21 @@ def get_mm_configs(
             kwarg_overrides,
             max_autotune,
         )
-        choices = []
-        # Third pass: Get adjusted choices and collect non-None ChoiceCaller objects
-        for ktc in adjusted_choices:
-            if ktc.choice is not None:
-                choices.append(ktc.choice)
-
-        return choices
+        # Layout optimization: if all choices are ExternKernelChoice and layout is FixedLayout, convert to FlexibleLayout
```

Comment:
This is typically the reverse order of how layout optimization goes. would it be possible to keep the layouts Flexible, then make them Fixed ?